solutions/cf-1359C.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- Case 1: pouring an equal number of cups of hot and cold water is equivalent to pouring one cup of each. Otherwise,
+- Case 2: there must be exactly one more cup of hot water poured than cold water poured. Thus, there will be $c_i + 1$ cups of hot water and $c_i$ cups of cold water for some $c \geq 0$.
+- It can be proven, using induction, that the average barrel temperatures when $c$ is odd is a monotonically decreasing function. Thus, we can binary search on the maximum number of odd cups $c$ which gives a temperature of at least $t$ by checking the odd integers around it.
+
+The answer will be either be the barrel with $2$, $c$, or $c + 2$ cups.
```

Comment:
```suggestion
- It can be proven, using induction, that the average barrel temperatures when $c_i$ is odd is a monotonically decreasing function. Thus, we can binary search on the maximum number of odd cups $c_i$ which gives a temperature of at least $t$ by checking the odd integers around it.

The answer will be either be the barrel with $2$, $c_i$, or $c_i + 2$ cups.
```